modules/core/src/main/scala/lucuma/odb/api/model/SpectroscopyRequirements.scala

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import cats.Eq
+import cats.syntax.all._
+import eu.timepit.refined._
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import io.circe.refined._
+import lucuma.odb.api.model.FocalPlane
+import lucuma.odb.api.model.SpectroscopyCapabilities
+import lucuma.core.math.Angle
+import lucuma.core.util.Display
+import lucuma.core.math.Wavelength
+import lucuma.core.util.Enumerated
+
+final case class SpectroscopyScienceRequirements(
+  wavelength:       Option[Wavelength],
+  resolution:       Option[PosInt],
+  signalToNoise:    Option[PosBigDecimal],
+  signalToNoiseAt:  Option[Wavelength],
+  wavelengthRange:  Option[Wavelength],
+  focalPlane:       Option[FocalPlane],
+  focalPlaneAngle:  Option[Angle],
+  capabilities:     Option[SpectroscopyCapabilities]
+)
+
+object SpectroscopyScienceRequirements {
+  val Default = SpectroscopyScienceRequirements(None, refineMV[Positive](1).some, None, None, None, None, None, None)
+
+  implicit val eqSpectroscopyConfigurationOptions: Eq[SpectroscopyScienceRequirements] = Eq.by(x =>
+    (x.wavelength,
+     x.resolution,
+     x.signalToNoise,
+     x.signalToNoiseAt,
+     x.wavelengthRange,
+     x.focalPlane,
+     x.focalPlaneAngle,
+     x.capabilities
+    )
+  )
+}
+
+object SpectroscopyScienceRequirementsModel {
+  final case class Input(
```

Comment:
I used this for Edit and Create. I'm not sure if this is correct. It seems that both patterns are used.

Reply:
"It depends." Often when you create something there are a lot of fields that are required.  If the user doesn't specify one or more required fields then you can't really create the `Foo` object.  When you edit though, the idea is to only specify the things you want to change.  Other stuff that goes unspecified in the editor is left untouched.

In this case, everything is optional upon create so this may be fine.  On the other hand, it leaves us with no way to distinguish between wanting to remove something and wanting to leave it untouched.  Say for example:

```
  "capabilities" : null
```

in the editor to remove them.  For that you need a separate editor and the field becomes a nullable clue `Input`.

Reply:
I see, it seems we want to support nulling a field